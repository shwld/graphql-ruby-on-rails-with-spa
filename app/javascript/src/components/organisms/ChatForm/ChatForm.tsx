import { ApolloError } from '@apollo/client/errors'
import React, { useState } from 'react'

import { Button, Card, Input } from '../../atoms'
import { FullErrorMessage } from '../../molecules/FullErrorMessage'

import { SayInput, useSayMutation } from '@/graphql/generated'
import { useCurrentUser } from '@/src/hooks/currentUser'

export const ChatFormTemplate: React.VFC<{
  error?: ApolloError
  children?: React.ReactNode
  submitting?: boolean
  className?: string
  onSubmit(input: SayInput): void
}> = ({ submitting, children, error, className = '', onSubmit }) => {
  const [text, setText] = useState<string>('')
  return (
    <form
      onSubmit={(e) => {
        e.preventDefault()
        onSubmit({ text })
      }}
      className={className}
    >
      <Card>
        {children}
        <p className="pt-10">Say!</p>
        <Input defaultValue={text} onChange={(e) => setText(e.target.value)} />
        <FullErrorMessage error={error} />
        <Button className="w-full mt-3" disabled={submitting}>
          Submit
        </Button>
      </Card>
    </form>
  )
}

export const ChatForm: React.VFC<{ className?: string }> = ({ className }) => {
  const { currentUser } = useCurrentUser()
  const [mutate, { error, loading }] = useSayMutation()

  if (!currentUser) return <></>
  return (
    <ChatFormTemplate
      onSubmit={(input) => {
        mutate({ variables: { input } })
      }}
      submitting={loading}
      error={error}
      className={className}
    >
      <p>This is example chat app!</p>
      Hello {currentUser ? currentUser.email : 'loading'}
      <p className="pt-10">Say!</p>
    </ChatFormTemplate>
  )
}
