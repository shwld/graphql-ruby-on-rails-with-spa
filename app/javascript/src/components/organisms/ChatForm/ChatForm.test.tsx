import React from 'react'
import { create } from 'react-test-renderer'

import { ChatFormTemplate } from './ChatForm'

describe('Organisms/ChatForm', () => {
  it('Snap Shot', () => {
    const component = create(<ChatFormTemplate onSubmit={jest.fn()} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
