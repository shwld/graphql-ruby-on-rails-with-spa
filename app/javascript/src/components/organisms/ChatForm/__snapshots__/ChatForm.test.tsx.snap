// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms/ChatForm Snap Shot 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="max-w-sm rounded overflow-hidden shadow-lg "
  >
    <div
      className="px-6 py-4"
    >
      <p
        className="pt-10"
      >
        Say!
      </p>
      <input
        className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline "
        defaultValue=""
        onChange={[Function]}
      />
      <button
        className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded w-full mt-3"
      >
        Submit
      </button>
    </div>
  </div>
</form>
`;
